### PR TITLE
ICU-22534 Promote almost all @draft ICU 73 APIs to @stable ICU 73

### PR DESCRIPTION
**Pull request by richgillam**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22534
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

API promotions spreadsheet: https://docs.google.com/spreadsheets/d/1fXcIjmG-BxcDktMWqi1QenB2CW592DVnPEdt_roWnuE/edit#gid=0

### CODE REVIEW

File: icu4c/source/i18n/unicode/simplenumberformatter.h [hunk 1]
```diff
@@ -94,7 +91,7 @@ class U_I18N_API SimpleNumber : public UMemory {
      *
      * This setting is applied upon formatting the number. 
      *
-     * @draft ICU 73
+     * @stable ICU 73
      */
     void setMinimumIntegerDigits(uint32_t minimumIntegerDigits, UErrorCode& status);
```
Comment (sffc) [inline icu4c/source/i18n/unicode/simplenumberformatter.h:96]: We have

```cpp
    /**
     * Truncates the most significant digits to the given maximum number of integer digits.
     *
     * This function immediately mutates the inner value.
     *
     * @draft ICU 73
     * @stable ICU 73
     */
    void truncateStart(uint32_t maximumIntegerDigits, UErrorCode& status);

    /**
     * Pads the beginning of the number with zeros up to the given minimum number of integer digits.
     *
     * This setting is applied upon formatting the number. 
     *
     * @draft ICU 73
     * @stable ICU 73
     */
    void setMinimumIntegerDigits(uint32_t minimumIntegerDigits, UErrorCode& status);
```

Why did we name the functions like this? I feel like they should either be

1. `setMaximumIntegerDigits` + `setMinimumIntegerDigits`
2. `truncateStart` + `padStart`

Comment (markusicu) [reply icu4c/source/i18n/unicode/simplenumberformatter.h:96]: @sffc can you please
1. propose what you think should be changed -- probably to align better with ICU4X / "newer convention"
2. comment here which functions should be held back at `@draft ICU 73` until you have a chance to change them (assuming the proposal goes through)?

File: icu4c/source/i18n/unicode/simplenumberformatter.h [hunk 2]
```diff
@@ -76,7 +73,7 @@ class U_I18N_API SimpleNumber : public UMemory {
      *
      * This function immediately mutates the inner value.
      *
-     * @draft ICU 73
+     * @stable ICU 73
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/simplenumberformatter.h:76]: please keep this one `@draft ICU 73` for now

File: icu4c/source/i18n/unicode/simplenumberformatter.h [hunk 3]
```diff
@@ -85,7 +82,7 @@ class U_I18N_API SimpleNumber : public UMemory {
...
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/simplenumberformatter.h:85]: please keep this one `@draft ICU 73` for now

File: icu4c/source/i18n/unicode/simplenumberformatter.h [hunk 4]
```diff
@@ -94,7 +91,7 @@ class U_I18N_API SimpleNumber : public UMemory {
      *
      * This setting is applied upon formatting the number. 
      *
-     * @draft ICU 73
+     * @stable ICU 73
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/simplenumberformatter.h:94]: please keep this one `@draft ICU 73` for now

File: icu4c/source/i18n/unicode/simplenumberformatter.h [hunk 5]
```diff
@@ -103,7 +100,7 @@ class U_I18N_API SimpleNumber : public UMemory {
      *
      * This setting is applied upon formatting the number.
      *
-     * @draft ICU 73
+     * @stable ICU 73
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/simplenumberformatter.h:103]: please keep this one `@draft ICU 73` for now

File: icu4c/source/i18n/unicode/usimplenumberformatter.h [hunk 6]
```diff
@@ -134,7 +131,7 @@ usnum_multiplyByPowerOfTen(USimpleNumber* unumber, int32_t power, UErrorCode* ec
  *
  * This function immediately mutates the inner value.
  *
- * @draft ICU 73
+ * @stable ICU 73
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/usimplenumberformatter.h:134]: please keep this one `@draft ICU 73` for now

File: icu4c/source/i18n/unicode/usimplenumberformatter.h [hunk 7]
```diff
@@ -145,7 +142,7 @@ usnum_roundTo(USimpleNumber* unumber, int32_t power, UNumberFormatRoundingMode r
  *
  * This setting is applied upon formatting the number.
  *
- * @draft ICU 73
+ * @stable ICU 73
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/usimplenumberformatter.h:145]: please keep this one `@draft ICU 73` for now

File: icu4c/source/i18n/unicode/usimplenumberformatter.h [hunk 8]
```diff
@@ -156,7 +153,7 @@ usnum_setMinimumIntegerDigits(USimpleNumber* unumber, int32_t minimumIntegerDigi
...
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/usimplenumberformatter.h:156]: please keep this one `@draft ICU 73` for now

File: icu4c/source/i18n/unicode/usimplenumberformatter.h [hunk 9]
```diff
@@ -167,7 +164,7 @@ usnum_setMinimumFractionDigits(USimpleNumber* unumber, int32_t minimumFractionDi
  *
  * This function immediately mutates the inner value.
  *
- * @draft ICU 73
+ * @stable ICU 73
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/usimplenumberformatter.h:167]: please keep this one `@draft ICU 73` for now

**Review by markusicu — CHANGES_REQUESTED** [review]

**Comment by markusicu** [comment]
Rich, please remember to add back the
```
#ifndef U_HIDE_DRAFT_API
...
#endif // U_HIDE_DRAFT_API
```
just around the two-times-four still-draft functions,
and squash.

**Comment by richgillam** [comment]
> Rich, please remember to add back

Oops.  Thanks for catching that.

**Comment by richgillam** [comment]
@markusicu I think I have this straightened out and squashed (after screwing something up and not knowing what I did).  PTAL.

**Review by markusicu — APPROVED** [review]
tnx

**Comment by FrankYFTang** [comment]
```
#ifdef U_HIDE_DRAFT_API
    // Must include all fields that will be in structs
    UCAL_FIELD_COUNT = UCAL_IS_LEAP_MONTH + 2,
#else  // U_HIDE_DRAFT_API (for UCAL_ORDINAL_MONTH)
    UCAL_FIELD_COUNT = UCAL_ORDINAL_MONTH + 1,
#endif  // U_HIDE_DRAFT_API (for UCAL_ORDINAL_MONTH)
```
this part also need to be changed to
```
    UCAL_FIELD_COUNT = UCAL_ORDINAL_MONTH + 1,
```

**Comment by FrankYFTang** [comment]
also
https://github.com/unicode-org/icu/blob/18c7d48b3e527ccead8ba93abd6059dfe1329258/icu4c/source/tools/toolutil/udbgutil.cpp#L110

**Comment by richgillam** [comment]
Thanks and apologies.  See https://github.com/unicode-org/icu/pull/2882

**Pull request merged**